edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubCertificates.cs

```diff
@@ -27,30 +35,67 @@ public class EdgeHubCertificates
         public static async Task<EdgeHubCertificates> LoadAsync(IConfigurationRoot configuration)
         {
             Preconditions.CheckNotNull(configuration, nameof(configuration));
-
+            string edgeHubDevCertPath = configuration.GetValue<string>(Constants.ConfigKey.EdgeHubDevServerCertificateFile);
+            string edgeHubDevPrivateKeyPath = configuration.GetValue<string>(Constants.ConfigKey.EdgeHubDevServerPrivateKeyFile);
+            string edgeHubDockerCertPFXPath = configuration.GetValue<string>(Constants.ConfigKey.EdgeHubServerCertificateFile);
+            string edgeHubDockerCaChainCertPath = configuration.GetValue<string>(Constants.ConfigKey.EdgeHubServerCAChainCertificateFile);
             string edgeHubConnectionString = configuration.GetValue<string>(Constants.ConfigKey.IotHubConnectionString);
 
-            // When connection string is not set it is edged mode
+            OperatingMode mode;
             if (string.IsNullOrEmpty(edgeHubConnectionString))
             {
-                var workloadUri = new Uri(configuration.GetValue<string>(Constants.ConfigKey.WorkloadUri));
-                string edgeHubHostname = configuration.GetValue<string>(Constants.ConfigKey.EdgeDeviceHostName);
-                string moduleId = configuration.GetValue<string>(Constants.ConfigKey.ModuleId);
-                string generationId = configuration.GetValue<string>(Constants.ConfigKey.ModuleGenerationId);
-                DateTime expiration = DateTime.UtcNow.AddDays(Constants.CertificateValidityDays);
-                (X509Certificate2 ServerCertificate, IEnumerable<X509Certificate2> CertificateChain) certificates =
-                    await CertificateHelper.GetServerCertificatesFromEdgelet(workloadUri, Constants.WorkloadApiVersion, moduleId, generationId, edgeHubHostname, expiration);
-
-                InstallCertificates(certificates.CertificateChain);
-                return new EdgeHubCertificates(certificates.ServerCertificate, certificates.CertificateChain?.ToList());
+                // When connection string is not set it is edged mode as iotedgd is expected to set this
+                mode = OperatingMode.IotEdged;
+            }
+            else if (!string.IsNullOrEmpty(edgeHubDevCertPath) &&
+                     !string.IsNullOrEmpty(edgeHubDevPrivateKeyPath))
+            {
+                mode = OperatingMode.IotEdgedDev;
             }
+            else if (!string.IsNullOrEmpty(edgeHubDockerCertPFXPath) &&
```

Comment:
There are three cases unfortunately its because of the certificate file format. This changed from as we moved to iotedged mode. So that is case 1. With iotedged file format we can either obtain this from the workload API or have a developer provide it for local dev/test. These are cases 2 and 3.

Reply:
Based on the fact that we have these scenarios called out above there was no choice but to handle the legacy certs and the iotedge workload style certificates.